src/Microsoft.SqlTools.ServiceLayer/TableDesigner/Contracts/DataModel/ObjectDataModelBase.cs

```diff
@@ -0,0 +1,20 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.ServiceLayer.TableDesigner.Contracts
+{
+    /// <summary>
+    /// The base class for data model object.
+    /// </summary>
+    public abstract class ObjectDataModelBase
+    {
+        public ObjectDataModelBase()
+        {
+            this.Name = new InputBoxProperties();
+        }
+
+        public InputBoxProperties Name { get; set; }
```

Comment:
`public InputBoxProperties Name { get; set; } = new InputBoxProperties();`

Although it seems a bit odd to have something as generic as "Name" actually be an input box. `NameInputBox` would be clearer...(or even have both since I would have expected name would just be a string you could access regardless...)

Reply:
not quite following this: ```Why isn't the model just the current values for the object that gets updated when a user selects a new one?```